airflow/utils/serve_logs.py

```diff
@@ -17,25 +17,61 @@
 
 """Serve logs process"""
 import os
+import time
 
-import flask
+from flask import Flask, abort, request, send_from_directory
+from itsdangerous import TimedJSONWebSignatureSerializer
 from setproctitle import setproctitle
 
 from airflow.configuration import conf
 
 
-def serve_logs():
-    """Serves logs generated by Worker"""
-    print("Starting flask")
-    flask_app = flask.Flask(__name__)
-    setproctitle("airflow serve-logs")
+def flask_app():
+    flask_app = Flask(__name__)
+    max_request_age = conf.getint('webserver', 'log_request_clock_grace', fallback=30)
+    log_directory = os.path.expanduser(conf.get('logging', 'BASE_LOG_FOLDER'))
+
+    signer = TimedJSONWebSignatureSerializer(
+        secret_key=conf.get('webserver', 'secret_key'),
```

Comment:
I think we should pass the salt parameter here as well to prevent the token from being used in the wrong context e.g. using a token from the getCode API in this method. This shouldn't work, but it's better to still explicitly prevent it.

Reply:
Do you mean something like adding `salt='logs'`?

Also probably doesn't hurt, but the formats are very different.

Reply:
Yes.  For now, we have used the tokens in two places, but in the future we may use them more often.  If we are unlucky, it may happen that it will be possible to generate a token that has a matching format. When we add salt, this token can be used here and no other place.

Reply:
For details, see: https://github.com/pallets/itsdangerous/blob/main/src/itsdangerous/signer.py#L78-L79

Reply:
Done in 27265516d